### ISSUE TITLE
[Feature Request] Support value_or_pointer arguments for oneMath functions

### ISSUE THREAD

**Issue by ndgrigorian** [oneapi-src/oneMKL#553]
OneAPI spec delineates [value_or_pointer](https://oneapi-spec.uxlfoundation.org/specifications/oneapi/v1.4-rev-1/elements/onemath/source/domains/blas/axpy#axpy-usm-version) which enables writing complex programs where the results of kernels can be chained together without bringing data to the host.

Providing these overloads would improve usability of oneMath.

**Comment by Rbiessy** [comment]
I agree it would be nice to implement this feature already in the spec. The spec uses `value_or_pointer` only for the BLAS domain and USM API, would that be enough for your purposes? @oneapi-src/onemkl-blas-write do you have any plan to work on that?

**Comment by ndgrigorian** [comment]
> I agree it would be nice to implement this feature already in the spec. The spec uses `value_or_pointer` only for the BLAS domain and USM API, would that be enough for your purposes? @oneapi-src/onemkl-blas-write do you have any plan to work on that?

Yes, it would be. It's also worth noting that rocBLAS and cuBLAS support pointers for scalar arguments.
